webview-ui/src/utils/transformation.ts

```diff
@@ -31,5 +31,25 @@ export const groupIncidentsByMsg = (
         acc[msg].push([uri, incident]);
         return acc;
       },
-      {} as { [msg: string]: [string, Incident][] },
+      {} as { [msg: string]: [string, EnhancedIncident][] },
     );
+
+export function enhanceIncidents(
+  incidents: Incident[] | undefined,
+  violation: EnhancedViolation | undefined,
+): EnhancedIncident[] {
+  if (!incidents || !violation) {
+    return [];
+  }
+
+  return incidents.map((incident) => ({
+    ...incident,
+    violationId: violation.id,
+    violationDescription: violation.description,
+    rulesetName: violation.rulesetName,
+    violationName: violation.violationName,
+    uri: incident.uri,
+    message: incident.message,
+    severity: incident.severity,
```

Comment:
This builds on what I said about considering moving this into `loadRulesets` where we have the rulesets + violations + incidents all available to us, but we must make sure that we fill in all of this correctly to give `kai` the best chance of providing meaningful solutions [here](https://github.com/konveyor/kai/blob/de6ca2ea455b9960dae5cded53112e18b1d24228/kai/analyzer_types.py#L89-L100).

```
    # copied it all here for ref
    ruleset_name: str
    ruleset_description: Optional[str] = None

    violation_name: str
    violation_description: Optional[str] = None
    violation_category: Category = Category.POTENTIAL
    violation_labels: list[str] = []
```